src/languages/clojure.js

```diff
@@ -45,20 +45,44 @@ export default function(hljs) {
       'lazy-seq spread list* str find-keyword keyword symbol gensym force rationalize'
   };
 
-  const SIMPLE_NUMBER_RE = '[-+]?\\d+(\\.\\d+)?';
-
   const SYMBOL = {
     begin: SYMBOL_RE,
     relevance: 0
   };
   const NUMBER = {
-    className: 'number',
-    begin: SIMPLE_NUMBER_RE,
-    relevance: 0
+    scope: 'number',
+    relevance: 0,
+    variants: [
+      {match: /[-+]?0[xX][0-9a-fA-F]+N?/},                           // hexadecimal                 // 0x2a
+      {match: /[-+]?0[0-7]+N?/},                                     // octal                       // 052
+      {match: /[-+]?[1-9][0-9]?[rR][0-9a-zA-Z]+N?/},                 // variable radix from 2 to 36 // 2r101010, 8r52, 36r16
+      {match: /[-+]?[0-9]+\/[0-9]+N?/},                              // ratio                       // 1/2
+      {match: /[-+]?[0-9]+((\.[0-9]*([eE][+-]?[0-9]+)?M?)|([eE][+-]?[0-9]+M?|M))/}, // float        // 0.42 4.2E-1M 42E1 42M
+      {match: /[-+]?([1-9][0-9]*|0)N?/},                             // int (don't match leading 0) // 42 42N
+    ]
   };
+  const CHARACTER = {
+    scope: 'character',
+    variants: [
+      {match: /\\o[0-3]?[0-7]{1,2}/},                             // Unicode Octal 0 - 377
+      {match: /\\u[0-9a-fA-F]{4}/},                               // Unicode Hex 0000 - FFFF
+      {match: /\\(newline|space|tab|formfeed|backspace|return)/}, // special characters
+      {match: /\\\S/}                                             // any non-whitespace char
+    ]
+  }
```

Comment:
@joshgoebel I refactored the Number and Character regex to be more maintainable.

Reply:
Now how would you write 2 - -3? Is there no confusion because it's lisp like?

Reply:
Clojure uses prefix notation, and `-` is actually a function like any other. There are no special operators symbols like in other languages. 

```clojure
(- 2 -3)
; or even more
(- 2 -3 +5) ; <- GitHub has broken highlighting. :-)
```

Reply:
What does that bottom one do?

Reply:
`(+ 1 2 3)` is like `1 + 2 + 3` in other languages. 

GitHub doesn't recognise numbers beginning with `+`. 

```clojure
42
-42
+42
```

I will send a PR to them as well. :-) While working on highlight.js's parsing, I started to notice errors and edgecases in other highlighters aswell.